### PR TITLE
fix(aws-amplify-react):Hide Forgot Password/Singup conditionally

### PR DESCRIPTION
**Pull request by yuth**
Users can pass a list of components to withAuthenticator and it would render only the passed list. If the user does not pass SignUp or Forgot password the SignIn form will not render the links
for SignUp and Forgot password

*Issue #, if available:*
Fixes issue #1467
*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/aws-amplify-react/src/Auth/SignIn.jsx [hunk 1]
```diff
@@ -106,9 +106,10 @@ export default class SignIn extends AuthPiece {
     }
 
     showComponent(theme) {
-        const { authState, hide, federated, onStateChange } = this.props;
+        const { authState, hide = [], federated, onStateChange } = this.props;
         if (hide && hide.includes(SignIn)) { return null; }
-
+        const hideSingUp = hide.some(component => component.name === 'SignUp')
```
Comment (powerful23) [inline packages/aws-amplify-react/src/Auth/SignIn.jsx]: hideSignUp

File: packages/aws-amplify-react/src/Auth/SignIn.jsx [hunk 2]
```diff
@@ -141,14 +142,18 @@ export default class SignIn extends AuthPiece {
...
+                                {I18n.get('Forgot Password')}
+                            </Link>
+                        }
                     </div>
                     <div style={Object.assign({textAlign: 'right'}, theme.col6)}>
-                        <Link theme={theme} onClick={() => this.changeState('signUp')}>
-                            {I18n.get('Sign Up')}
-                        </Link>
+                        {
+                            !hideSingUp && <Link theme={theme} onClick={() => this.changeState('signUp')}>
```
Comment (powerful23) [inline packages/aws-amplify-react/src/Auth/SignIn.jsx]: hideSignUp

**Comment by mogadanez** [comment]
reverted back in https://github.com/aws-amplify/amplify-js/commit/41d3184239b4087d750c86ae7401843afe6f6555#diff-b7346e291e605cfa8336307c89af06d2

**Comment by Statsministeriet** [comment]
You can use the hideSignUp 
`      <AmplifyAuthenticator usernameAlias="email">
        <AmplifySignIn slot="sign-in" usernameAlias="email" hideSignUp={true}  handleAuthStateChange={handleAuthState}/>
      </AmplifyAuthenticator>
`

**Pull request merged**